### PR TITLE
Simple Classic: Upsell pages - Handle redirection after activation

### PR DESCRIPTION
**Pull request by miksansegundo**
Related to https://github.com/Automattic/dotcom-forge/issues/7492 D150471-code

## Proposed Changes
* Use the URL param `redirect_to` to handle redirection after the activation
* Default to `/hosting-config/{ SITE ID }`

>[!Note]
> I'll follow up with the ⚙️ diff to pass the redirection URL from the upsell pages

## Why are these changes being made?

* To redirect users to the feature after the activation via the upsell pages 

## Testing Instructions

* Prepare a simple site with the Creator plan
* Go to `wordpress.com/dev-tools/{ SITE ID }/?redirect_to={ ANY URL }`
* Click on "Activate now"
* Verify you are redirected to that URL after the activation

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/dev-tools/components/dev-tools.tsx
Comment (mmtr) [inline client/dev-tools/components/dev-tools.tsx:41]: Why is `useRef` needed here?

Comment (fushar) [reply client/dev-tools/components/dev-tools.tsx:41]: I think I know why... it is because `redirect_to` is not included here, so the param will be gone as soon as it is mounted.

https://github.com/Automattic/wp-calypso/blob/c640cff1674b46c767f19a31e54358d1822507ad/client/sites-dashboard-v2/controller.tsx#L19-L26

A simpler solution is maybe to add that query param in the above logic 🤔

Comment (miksansegundo) [reply client/dev-tools/components/dev-tools.tsx:41]: Another solution is get the URL param out of the functional component to avoid losing it after renders.

Why not just use the ref?

Comment (fushar) [reply client/dev-tools/components/dev-tools.tsx:41]: > Why not just use the ref?

I don't have strong feeling against it. Just it could confuse future developers as it's not that obvious why it's needed here. If we want to use it, I suggest we add some code comment 🤔

Comment (miksansegundo) [reply client/dev-tools/components/dev-tools.tsx:41]: I tried adding the param `redirect_to` to `getQueryParams` but it didn't work, so I'm adding a comment to keep the ref.

File: client/dev-tools/components/dev-tools.tsx
Comment (arthur791004) [inline client/dev-tools/components/dev-tools.tsx]: do you want to show the modal at the beginning by checking the query parameters?

**Review by fushar — APPROVED** [review]
Works for me 👍

File: client/dev-tools/components/dev-tools.tsx
Comment (fushar) [reply client/dev-tools/components/dev-tools.tsx]: Looks it's what this PR https://github.com/Automattic/wp-calypso/pull/91226 does.

**Comment by miksansegundo** [comment]
D150471-code is ready for testing the links from the upsell pages.

I have also rebased this PR to get the changes that use the param `activate` to open the modal.

**Comment by miksansegundo** [comment]
I tested with the diff, and it works fine. I'll wait for the diff review before merging this PR.

https://github.com/Automattic/wp-calypso/assets/1881481/92f8ae6a-102c-432a-931b-3f2ada29a96a

**Review by arthur791004 — APPROVED** [review]

**Pull request merged**
